cdap-app-fabric/src/main/java/co/cask/cdap/gateway/handlers/ArtifactHttpHandler.java

```diff
@@ -699,6 +698,10 @@ public BodyConsumer addArtifact(HttpRequest request, HttpResponder responder,
       additionalPluginClasses = ImmutableSet.of();
     } else {
       try {
+        // verify that all the plugin classes are valid
+        for (JsonElement jsonElement : GSON.fromJson(pluginClasses, JsonElement.class).getAsJsonArray()) {
+          PluginClass.verifyJSON(jsonElement.getAsJsonObject());
```

Comment:
The idea was to follow something like the ProfileCreateRequest, where you still use Gson to deserialize the pluginClasses into a `Set<PluginClass>`, but then call validate() on each PluginClass, which should make sure any fields that are required are present, etc. We want to avoid dealing with JsonElements whenever possible.